### DISCUSSION TITLE
Get rewarded ads coins worth value in USD?

### DISCUSSION THREAD

**Discussion by rgouzal** [invertase/react-native-firebase#6320] [Q&A] [answered]
I am looking  for a way where to get rewarded ads coins and use it as a way of discount on customer checkout, so customer watch ads collect coins in some days and based on how much they collected a discount is applied but the issue is I can't find a documentation that shows how to get worth value in USD or any other currency for the coins that was rewarded. Is there any work around? I have seen some mobile apps that have that feature already integrated so I am wondering that it is possible.

**Comment by mikehardy** [comment, marked as answer]
I think you want react-native-google-mobile-ads - https://github.com/invertase/react-native-google-mobile-ads - you show a rewarded ad and *you* set the value of the reward, track that (on the server, perhaps with `@react-native-firebase/firestore` ? and go from there ?

**Reply by rgouzal** [reply]
@mikehardy You mean Reward amount can be set? where?
 I meant from what mentioned here https://docs.page/invertase/react-native-google-mobile-ads/displaying-ads#rewarded-ads when doing:

```
    const unsubscribeEarned = rewarded.addAdEventListener(
      RewardedAdEventType.EARNED_REWARD,
      reward => {
        console.log('User earned reward of ', reward);
      },
    );
```

The reward object will actually have amount of coins rewarded, for example I think some ads when its showed up it gets always 10 coins which I think its not the same real worth value in USD for the ad shown. How can I convert that coins to USD currency based value?

**Reply by mikehardy** [reply]
@rgouzal this is not specific to this module, you'll need to read how to do this from the upstream documentation, for example 

https://support.google.com/admanager/answer/7386053?hl=en 
https://support.google.com/admob/answer/7311747?hl=en 
https://www.blog.google/products/admob/beginners-guide-to-admob-rewarded-ads/ 

*you* decide the value, *you* decide how to verify/track it

**Reply by rgouzal** [reply]
@mikehardy Thanks for explaining more in details, I was confused of how to design a rewarding system but I think I understand now. How can I estimate profit of 1 full video ad for a user tough? Is there any source or page has profit estimation?
My idea is I want users to be able use points or rewards to purchase items or to cash them out.

**Reply by mikehardy** [reply]
With apologies this is out of the scope of this module, we're here to make sure you can *use* the google mobile ads SDKs in react-native but the "how" and the "why is best handled with upstream documentation + videos etc
